pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,72 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def _complete_unitary(columns):
+    """
+    Completes a unitary matrix given a list of orthonormal columns.
+
+    Args:
+        columns (List[Array]): List of initial orthonormal columns of dimension d.
+
+    Returns:
+        Array: Completed unitary matrix of dimension :math:`(d, d)`.
+    """
+
+    columns = qml.math.stack(columns).T
+    d = columns.shape[0]
+    k = columns.shape[1]
+
+    unitary = qml.math.zeros_like(columns @ columns.T)
+
+    if qml.math.get_interface(columns) == "jax":
+        unitary = unitary.at[:, :k].set(columns)
+
+    else:
+        unitary[:, :k] = columns
+
+    # Complete the remaining columns using Gram-Schmidt
+    np.random.seed(42)
```

Comment:
Global seeds have far rearching consequences and will interfere with other parts of parts of the workflow and the users code and choices.  Can we use a local random generator and give the user control over the seed?

Reply:
let me change this part 👍

Reply:
Updated to this:

```python
rng = np.random.default_rng(42)
     for j in range(k, d):
         random_vec = qml.math.array(rng.random(d))
```

Reply:
Thanks! I've dismissed my review.